src/healthkit.cs

```diff
@@ -492,6 +504,16 @@ interface HKHealthStore {
 		[Notification]
 		[Field ("HKUserPreferencesDidChangeNotification")]
 		NSString UserPreferencesDidChangeNotification { get; }
+
+		[Async]
+		[Watch (5,0), iOS (12,0)]
+		[Export ("getRequestStatusForAuthorizationToShareTypes:readTypes:completion:")]
+		void GetRequestStatus (NSSet<HKSampleType> typesToShare, NSSet<HKObjectType> typesToRead, Action<HKAuthorizationRequestStatus, NSError> completion);
```

Comment:
The method for `requestAuthorizationToShareTypes:readTypes:completion:` is named `RequestAuthorizationToShare`, so I think this would be better named as `GetRequestStatusForAuthorizationToShare`.

Another argument for the same change is that the For* suffix does not refer to the first parameter, but explains what the function does.